packit_service/worker/handlers/vm_image.py

```diff
@@ -0,0 +1,139 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+"""
+This file defines classes for job handlers specific for Fedmsg events
+"""
+
+import logging
+from typing import Tuple, Type
+
+from packit.config import (
+    JobType,
+)
+from packit_service.worker.checker.abstract import Checker
+from packit_service.worker.events import (
+    AbstractPRCommentEvent,
+    VMImageBuildResultEvent,
+)
+from packit_service.worker.handlers.abstract import (
+    JobHandler,
+    RetriableJobHandler,
+    TaskName,
+    configured_as,
+    reacts_to,
+    run_for_comment,
+)
+from packit_service.worker.result import TaskResults
+from packit_service.worker.checker.vm_image import (
+    IsCoprBuildForChrootOk,
+    GetVMImageBuildReporterFromJobHelperMixin,
+)
+from packit_service.worker.mixin import (
+    ConfigFromEventMixin,
+    GetVMImageBuilderMixin,
+)
+from packit_service.models import (
+    VMImageBuildTargetModel,
+    VMImageBuildStatus,
+    PipelineModel,
+)
+
+from packit_service.celerizer import celery_app
+
+logger = logging.getLogger(__name__)
+
+
+@configured_as(job_type=JobType.vm_image_build)
+@run_for_comment(command="vm-image-build")
+@reacts_to(AbstractPRCommentEvent)
+class VMImageBuildHandler(
+    RetriableJobHandler,
+    ConfigFromEventMixin,
+    GetVMImageBuilderMixin,
+    GetVMImageBuildReporterFromJobHelperMixin,
+):
+    task_name = TaskName.vm_image_build
+
+    @staticmethod
+    def get_checkers() -> Tuple[Type[Checker], ...]:
+        return (IsCoprBuildForChrootOk,)
+
+    def run(self) -> TaskResults:
+        if not self.job_config:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": f"Job configuration not found for project {self.project.repo}"
+                },
+            )
+
+        image_id = self.vm_image_builder.create_image(
+            self.image_distribution,
+            self.image_name,
+            self.image_request,
+            self.image_customizations,
+            self.project_url,
+        )
+
+        run_model = PipelineModel.create(
+            type=self.data.db_trigger.job_trigger_model_type,
+            trigger_id=self.data.db_trigger.id,
+        )
+        VMImageBuildTargetModel.create(
+            build_id=image_id,
+            commit_sha=self.data.commit_sha,
+            project_name=self.project_name,
+            owner=self.owner,
+            project_url=self.project_url,
+            target=self.chroot,
+            status=VMImageBuildStatus.pending,
+            run_model=run_model,
+        )
+
+        self.report_status(VMImageBuildStatus.pending, "")
+
+        celery_app.send_task(
+            "task.babysit_vm_image_build",
+            args=(image_id,),
+            countdown=2,  # do the first check in 120s
```

Comment:
I recall that when my image build request was wrong, my build failed pretty quickly so can we start in 10s for example?